be/models.py

```diff
@@ -48,34 +49,81 @@ class SearchParams(BaseModel):
+
+        class AggregationResponse(BaseModel):
+            __annotations__ = {
+                name: Aggregation
+                for name, (_, filterable) in fields.items()
+                if filterable
+            }
+
+        class SearchParamsExtended(SearchParams):
+            # Define filterable fields with default values
+            locals().update(
+                {
+                    name: Field(None, description=f"{name} query", alias=name)
+                    for name, (type_, filterable) in fields.items()
+                    if filterable
+                }
+            )
+            __annotations__ = {
+                name: type_ | None
+                for name, (type_, filterable) in fields.items()
+                if filterable
+            }
+            # Define default sort field and order.
+            sort_field: str | None = Field(
+                self.default_sort_field, description="Sort field"
+            )
+            sort_order: Literal["desc", "asc"] = Field(
+                self.default_sort_order, description="Sort order"
+            )
+
+        return Data, AggregationResponse, SearchParamsExtended
 
-class MaveDBData(BaseModel):
-    urn: str
-    title: str
-    shortDescription: str
-    sequenceType: str
-    geneName: str
-    geneCategory: str
-    publicationUrl: str
-    # TODO: undef values are strings, convert it to None
-    publicationYear: int|str
-    numVariants: int|str
-
-class MaveDBAggregationResponse(BaseModel):
-    publicationYear: Aggregation
-    sequenceType: Aggregation
-    geneCategory: Aggregation
-
-class MaveDBSearchParams(SearchParams):
-    sort_field: str | None = Field("publicationYear", description="Sort field")
-    sort_order: Literal["desc", "asc"] = "desc"
-    publication_year: str | None = Field(
-        None, description="PublicationYear query", alias="publicationYear"
-    )
-    gene_category: str | None = Field(
-        None, description="GeneCategory query", alias="geneCategory"
-    )
-    sequence_type: str | None = Field(
-        None, description="SequenceType query", alias="sequenceType"
-    )
+
+# MaveDB.
+mavedb = DataSource(
+    name="MaveDB",
+    fields=[
+        FieldDefinition(name="urn", type=str),
+        FieldDefinition(name="title", type=str),
+        FieldDefinition(name="shortDescription", type=str),
+        FieldDefinition(name="sequenceType", type=str, filterable=True),
+        FieldDefinition(name="geneName", type=str),
+        FieldDefinition(name="geneCategory", type=str, filterable=True),
+        FieldDefinition(name="publicationUrl", type=str),
+        FieldDefinition(name="publicationYear", type=int, filterable=True),
+        FieldDefinition(name="numVariants", type=int),
+    ],
+    default_sort_field="publicationYear",
+    default_sort_order="desc",
+)
+MaveDBData, MaveDBAggregationResponse, MaveDBSearchParams = mavedb.generate_classes()
```

Comment:
this line is longer 80 chars but I guess all formatting issues are fixed in another PR, right?

Reply:
Actually the limit used by the Black code style is 88 characters (they've done some research into the optimal value and have a good explanation [here](https://black.readthedocs.io/en/stable/the_black_code_style/current_style.html#line-length)), but yes, in any case all reformatting will be done in the subsequent PR